teos/src/main.rs

```diff
@@ -193,7 +190,7 @@ async fn main() {
             .validate(block_hash)
             .unwrap()
```

Comment:
Ummm, doesn't this fail as well when we are out of blocks?

Reply:
Actually, it does not because the headers chain is preserved even if the blocks are pruned.